apis/Google.Devtools.AspNet/Google.Devtools.AspNet.Tests/CloudTrace/BufferingTraceConsumerTest.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Devtools.Cloudtrace.V1;
+using Moq;
+using Xunit;
+
+namespace Google.Devtools.AspNet.Tests.CloudTrace
+{
+    public class BufferingTraceConsumerTest
+    {
+        private Trace CreateTrace()
+        {
+            return new Trace()
+            {
+                ProjectId = "some-pid",
+                TraceId = "f05445aa7843bc8bf206b12000100000"
+            };
+        }
+
+        [Fact]
+        public void Receive()
+        {
+            int bufferSize = 1024;
+            Traces traces = new Traces();
+            traces.Traces_.Add(CreateTrace());
+
+            Mock<ITraceConsumer> mockConsumer = new Mock<ITraceConsumer>();
+            BufferingTraceConsumer consumer = BufferingTraceConsumer.Create(mockConsumer.Object, bufferSize);
+            consumer.Receive(traces);
+            mockConsumer.Verify(c => c.Receive(It.IsAny<Traces>()), Times.Never());
+
+            traces = new Traces();
+            while (traces.Traces_.Count < 1000 && traces.CalculateSize() < bufferSize)
+            {
+                traces.Traces_.Add(CreateTrace());
+            }
+
+            // Add the initial trace to the current list.
+            Traces combined = traces;
```

Comment:
I don't see why we have a different variable here. Why not just `traces.Traces_.Add(CreateTrace());`? I'm a bit confused as to exactly what we're testing here, to be honest - I think the fact that the `traces` variable changes value doesn't help here either.